x/gov/keeper/msg_server.go

```diff
@@ -168,5 +175,84 @@ func (k msgServer) Deposit(goCtx context.Context, msg *v1beta1.MsgDeposit) (*v1b
 		)
 	}
 
+	return &types.MsgDepositResponse{}, nil
+}
+
+type legacyMsgServer struct {
+	Server types.MsgServer
+}
+
+// NewLegacyMsgServerImpl returns an implementation of the v1beta1 legacy MsgServer interface. It wraps around
+// the current MsgServer
+func NewLegacyMsgServerImpl(keeper Keeper) v1beta1.MsgServer {
+	return &legacyMsgServer{Server: NewMsgServerImpl(keeper)}
```

Comment:
We're creating another `NewMsgServerImpl(keeper)` here. Can we use the same instance as above?

Reply:
Yup of course. I just liked that they had the same argument